src/overmap.cpp

```diff
@@ -1229,6 +1238,11 @@ oter_id &overmap::ter(const int x, const int y, const int z)
     return layer[z + OVERMAP_DEPTH].terrain[x][y];
 }
 
+oter_id &overmap::ter( const tripoint &p )
```

Comment:
Agreed, but I didn't invent this approach here, merely added an overloaded version of the function. I guess, references could improve performance under certain conditions, since both getting and setting the value perform the same steps which involve checking bounds and calculating indexes. However, the impact is probably unnoticeable.